lib/rbUtil.js

```diff
@@ -239,15 +239,19 @@ function HTTPError(response) {
     if (response.body && response.body.type) {
         this.message += ': ' + response.body.type;
     }
-
-    for (var key in response) {
-        this[key] = response[key];
-    }
+    Object.assign(this, response);
 }
 util.inherits(HTTPError, Error);
 
 rbUtil.HTTPError = HTTPError;
 
+function HTTPRedirect(response) {
```

Comment:
I'm a bit torn on treating HTTPRedirect as an error. It might be convenient to short-cut the processing this way, but also assumes that code several layers deep knows about how a response is going to be used.